app-agent/src/exec_process.rs

```diff
@@ -33,6 +56,165 @@ pub fn exec_child(external_command: String, args: Vec<String>) -> anyhow::Result
+            let path = Path::new(&external_command);
+            let external_command_corrected_name: String;
+            if path.is_absolute() {
+                external_command_corrected_name = path
+                    .file_name()
+                    .and_then(|os_str| os_str.to_str().map(|s| s.to_string()))
+                    .unwrap_or_else(|| external_command.to_string());
+            } else {
+                external_command_corrected_name = external_command
+                    .strip_prefix("./")
+                    .unwrap_or(&external_command)
+                    .to_string();
+            }
+
+            let distance = super::word_distance::distance_with_adjacent_transposition(
+                &external_command_corrected_name.clone(),
+                &entry_string.clone(),
+            );
+            if distance < 3 && distance < lowest_distance {
+                lowest_distance = distance;
+                best_element = Some((entry_string, distance));
+            }
+        }
+    }
+    match best_element {
+        Some((element, distance)) => {
+            let argument_list = args
+                .iter()
+                .map(|arg| {
+                    if arg.contains(' ') {
+                        format!("\"{}\"", arg)
+                    } else {
+                        arg.to_string()
+                    }
+                })
+                .collect::<Vec<_>>()
+                .join(" ");
+            if distance == 0 {
+                log::info!(
+                    "💡 Hint: A file named '{}' exists in the current directory. Prepend ./ to execute it.",
+                    element
+                );
+                log::info!("Example: {} exec ./{} {}", app_path, element, argument_list);
+            } else {
+                log::info!("💡 Hint: Did you mean ./{} {}", element, argument_list);
+            }
+            return "File not found but another appears to match".to_string();
+        }
+        None => {
+            log::warn!(
+                "💡 Hint: No matching file exists in the current directory. Please try again we a different one."
+            );
+        }
+    }
+    "Sorry but the file was not found".to_string()
+}
+
+fn check_missing_permissions(current_permissions: u32, required_permissions: u32) -> u32 {
+    required_permissions & !current_permissions
+}
+
+fn find_a_parent_with_perm_issue(path: String) -> anyhow::Result<std::path::PathBuf, String> {
+    // Current parent can change if a parent of the parent don't have the correct rights
+    let mut current_parent = match std::path::Path::new(&path).parent() {
+        Some(parent) => parent,
+        None => return Err("".to_string()),
+    };
+    // Through this loop I will iterate over parent of parent until I can retrieve metadata, it will show the first folder
+    // that I can't execute and suggest to the user to grant execution rights.
+    loop {
```

Comment:
Isn't this a bit dangerous? We can make it explode with very long paths. Maybe we should just give up and let the user figure it out? Or at least put a loop limit?